packages/flutter/lib/src/rendering/sliver.dart

```diff
@@ -369,32 +369,42 @@ class SliverConstraints extends Constraints {
         && remainingPaintExtent >= 0.0;
   }
 
-  /// Returns [BoxConstraints] that reflects the sliver constraints.
+  /// Returns [BoxConstraints] that reflect the sliver constraints.
   ///
   /// The `minExtent` and `maxExtent` are used as the constraints in the main
-  /// axis. If non-null, the given `crossAxisExtent` is used as a tight
-  /// constraint in the cross axis. Otherwise, the [crossAxisExtent] from this
-  /// object is used as a constraint in the cross axis.
+  /// axis. If non-null, and the given `crossAxisExtent` is finite, it is used
+  /// as a tight constraint in the cross axis. When [double.infinity], the
+  /// constraints will be loose in the cross axis. If the provided
+  /// `crossAxisExtent` is null, the [crossAxisExtent] from this object is used
+  /// as a tight constraint in the cross axis.
   ///
   /// Useful for slivers that have [RenderBox] children.
   BoxConstraints asBoxConstraints({
```

Comment:
The changes to this function are also new here. You will want to update your branches with this as well.